src/main/java/com/uid2/admin/audit/QLDBAuditMiddleware.java

```diff
@@ -0,0 +1,58 @@
+package com.uid2.admin.audit;
+
+import com.uid2.admin.Constants;
+import com.uid2.admin.vertx.service.IService;
+import com.uid2.shared.auth.IAuthorizable;
+import io.vertx.core.Handler;
+import io.vertx.ext.web.RoutingContext;
+
+import java.time.Instant;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Function;
+
+public class QLDBAuditMiddleware implements AuditMiddleware{
+    private final AuditWriter auditWriter;
+
+    protected QLDBAuditMiddleware(AuditWriter writer){
```

Comment:
Now middleware is agnostic to audit storage. Better just call it AuditMIddleware

Reply:
Good point. AuditMiddlewareImpl

Reply:
AuditMiddleware is already an interface